pkg/experiment/sensitivity/workload_factory.go

```diff
@@ -0,0 +1,276 @@
+		"Number of L1 instruction cache best effort processes to be run",
+		l1iDefaultProcessNumber,
+	)
+
+	// L3ProcessNumber represents number of L3 data cache aggressor processes to be run
+	L3ProcessNumber = conf.NewIntFlag(
+		"experiment_be_l3_processes_number",
+		"Number of L3 data cache best effort processes to be run",
+		l3DefaultProcessNumber,
+	)
+
+	// MembwProcessNumber represents number of membw aggressor processes to be run
+	MembwProcessNumber = conf.NewIntFlag(
+		"experiment_be_membw_processes_number",
+		"Number of membw best effort processes to be run",
+		membwDefaultProcessNumber,
+	)
+)
+
+// WorkloadFactory is creator for High Priority and Best Effort workloads with
+// default or custom isolation.
+type WorkloadFactory struct {
+	executorFactory ExecutorFactory
+
+	hpIsolation isolation.Decorator
+	l1Isolation isolation.Decorator
+	l3Isolation isolation.Decorator
+}
+
+// NewDefaultWorkloadFactory returns factory that would create Workloads on Kubernetes executor
+// or Local executor, depending on flags.
+func NewDefaultWorkloadFactory() WorkloadFactory {
+	return NewWorkloadFactory(NewExecutorFactory())
+}
+
+// NewWorkloadFactory creates new instance of WorkloadFactory.
+func NewWorkloadFactory(executorFactory ExecutorFactory) WorkloadFactory {
+	hpIsolation, l1Isolation, l3Isolation := GetWorkloadsIsolations()
+
+	return NewWorkloadFactoryWithIsolation(executorFactory, hpIsolation, l1Isolation, l3Isolation)
+}
+
+// NewWorkloadFactoryWithIsolation creates new workload factory with custom isolation.
+func NewWorkloadFactoryWithIsolation(
+	factory ExecutorFactory,
+	hpIsolation isolation.Decorator,
+	l1Isolation isolation.Decorator,
+	l3Isolation isolation.Decorator) WorkloadFactory {
+	return WorkloadFactory{
+		executorFactory: factory,
+		hpIsolation:     hpIsolation,
+		l1Isolation:     l1Isolation,
+		l3Isolation:     l3Isolation,
+	}
+}
+
+// BuildDefaultHighPriorityLauncher builds High Priority workload launcher with predefined isolation.
+func (factory *WorkloadFactory) BuildDefaultHighPriorityLauncher(
+	workloadName string, tags snap.Tags) (launcher executor.Launcher, err error) {
+	return factory.BuildHighPriorityLauncherWithIsolation(workloadName, factory.hpIsolation, tags)
+}
+
+// BuildHighPriorityLauncherWithIsolation builds High Priority launcher with provided isolation.
+func (factory *WorkloadFactory) BuildHighPriorityLauncherWithIsolation(
+	workloadName string,
+	decorators isolation.Decorator,
+	tags snap.Tags) (launcher executor.Launcher, err error) {
+	return factory.createHighPriorityWorkload(workloadName, decorators, tags)
+}
+
+// BuildDefaultBestEffortLauncher builds Best Effort workload launcher with predefined isolation.
+func (factory *WorkloadFactory) BuildDefaultBestEffortLauncher(
+	workloadName string,
+	tags snap.Tags) (launcher executor.Launcher, err error) {
+	return factory.BuildBestEffortLauncherWithIsolation(workloadName, factory.getDefaultBestEffortIsolation(workloadName), tags)
```

Comment:
@skonefal - you can call `createBestEffortWorkload()` directly, omitting `BuildBestEffortLauncherWithIsolation()` that adds no logic.